### PR TITLE
TEMP: add more logging to debug password reset issues. 🐿 v2.10.2

### PR DESCRIPTION
**Pull request by AbdullahAli**

### CODE REVIEW

File: app/util/substitutionData.js [hunk 1]
```diff
@@ -1,15 +1,32 @@
...
+const rawLogger = require('../../logger');
+
+// || '' to be on the safe side and prevent split being called on null/undefined error if env var is not set momentarily
+const USERS_HAVING_PASSWORD_ISSUES = (process.env.USERS_HAVING_PASSWORD_ISSUES || '').split(' ');
 
 function merge(users, subData = [], adsData = []) {
 	const normalizedRecipients = {};
+
 	users.forEach(user => {
+		if (USERS_HAVING_PASSWORD_ISSUES.includes(user.email.toLowerCase())) {
```
Comment (keirog) [inline app/util/substitutionData.js]: Probably good to check for the existence of `user.email` too, especially as we’re seeing those `.toLowerCase` errors

File: app/util/substitutionData.js [hunk 2]
```diff
@@ -1,15 +1,32 @@
...
+			rawLogger.verbose({ event: 'USER_WITH_PASSWORD_RESET_ISSUE', stage: 'NORMALISING_USER', userEmail: user.email, userUUID: user.uuid, user });
+		}
+
 		normalizedRecipients[user.email] = user;
 		normalizedRecipients[user.uuid] = user;
 	});
 
 	subData.forEach((subData) => {
+		if (USERS_HAVING_PASSWORD_ISSUES.includes(subData.email.toLowerCase())) {
```
Comment (keirog) [inline app/util/substitutionData.js]: Same here

File: app/util/substitutionData.js [hunk 3]
```diff
@@ -1,15 +1,32 @@
...
+			rawLogger.verbose({ event: 'USER_WITH_PASSWORD_RESET_ISSUE', stage: 'SUBTITUTING_USER', userEmail: subData.email.toLowerCase(), userUUID: subData.uuid, subData });
+		}
+
 		const user = normalizedRecipients[subData.email.toLowerCase()] || normalizedRecipients[subData.uuid];
 		assign(user, subData);
+
+		if (USERS_HAVING_PASSWORD_ISSUES.includes(subData.email.toLowerCase())) {
```
Comment (keirog) [inline app/util/substitutionData.js]: + here

**Comment by AbdullahAli** [comment]
@keirog refactored 👍🏼 can you take another look please

**Pull request merged**